### PR TITLE
[S27] callable for input

### PR DESCRIPTION
**Pull request by saanobhaai**

### CODE REVIEW

**Comment by saanobhaai** [comment]
[SCL zonal stats task](https://trello.com/c/naAEHnHp/27-scl-zonal-stats-task)

File: task_base.py [hunk 1]
```diff
@@ -17,6 +17,14 @@ class Task(object):
     inputs = {}
     wait_for_outputs = True
 
+    def _set_inputs(self):
+        for input_key, i in self.inputs.items():
+            for key, val in i.items():
+                if isinstance(val, str) and hasattr(self.__class__, val):
```
Comment (gridcell) [inline task_base.py]: instead of nesting if statements do the opposite and use `continue`

File: task_base.py [hunk 2]
```diff
@@ -86,13 +96,17 @@ class EETask(GeoTask):
...
+    def _create_ee_path(asset_path, ic=False):
         path_segments = asset_path.split('/')
         # first two segments are user/project root (e.g. projects/HII)
-        for i in range(2, len(path_segments)):
+        path_length = len(path_segments)
+        for i in range(2, path_length):
             path = '/'.join(path_segments[:i + 1])
             if not ee.data.getInfo(path):
-                ee.data.createAsset({'type': 'Folder'}, opt_path=path)
+                if i == path_length - 1 and ic:
```
Comment (gridcell) [inline task_base.py]: same nesting comment as above

File: task_base.py [hunk 3]
```diff
@@ -86,13 +96,17 @@ class EETask(GeoTask):
     EEDATATYPES = [IMAGECOLLECTION, FEATURECOLLECTION, IMAGE]
 
     @staticmethod
-    def _create_ee_path(asset_path):
+    def _create_ee_path(asset_path, ic=False):
```
Comment (gridcell) [inline task_base.py]: ic should be spelled out as an arg variable, so it's clear what it is.  also, should a static method be private

File: task_base.py [hunk 4]
```diff
@@ -172,11 +186,11 @@ def flatten_inputs(self):
                 return_properties[key] = propval
         return return_properties
 
-    def export_image_ee(self, image, asset_path):
+    def export_image_ee(self, image, asset_path, ic=True):
```
Comment (gridcell) [inline task_base.py]: should be `image_collection` instead of `ic`

**Pull request merged**